File: youlexuan-sellergoods-interface/src/main/java/com/offcn/sellergoods/service/BrandService.java
package com.offcn.sellergoods.service;

import com.offcn.entity.PageResult;
import com.offcn.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {
    public List<TbBrand> findAll();
    public PageResult findPage(int pageNum,int pageSize);
    public void add(TbBrand brand);
    public TbBrand findOne(Long id);
    public void update(TbBrand brand);
    public void delete(Long[] ids);
    public PageResult findPage(TbBrand brand,int pageNum,int pageSize);
    List<Map> selectOptionList();
}

File: youlexuan-manager-web/src/main/webapp/js/service/brandService.js
//定义Service层
app.service('brandService',function($http){
    //定义方法读取列表数据
    this.findAll=function(){
        return $http.get('../brand/findAll.do');
    }
    //获取分页数据
    this.findPage=function(page,rows){
        return $http.get('../brand/findPage.do?page='+page+'&rows='+rows);
    }
    //保存
    this.add=function(entity){
        return $http.post('../brand/add.do',entity);
    }
    //修改
    this.update=function(entity){
        return $http.post('../brand/update.do',entity);
    }
    //删除
    this.dele=function(ids){
        return $http.get('../brand/delete.do?ids='+ids);
    }
    //获取制定id品牌信息
    this.findOne=function(id){
        return $http.get('../brand/findOne.do?id='+id);
    }
    //搜索
    this.search=function(page,rows,searchEntity){

        return $http.post('../brand/search.do?page='+page+"&rows="+rows,searchEntity);
    }

    //下拉列表数据
    this.selectOptionList=function(){
        return $http.get('../brand/selectOptionList.do');
    }
});